book/src/platformer/05_platformer_physics.md

```diff
@@ -0,0 +1,244 @@
+# Platforming physics
+
+Designing the physics system for a platformer can be as complicated as you want it to be.
+Modern platformers use a vast set of tricks to make it feel responsive and fair.
+We're going to ignore it all to make something simple.
+
+This will be the character we use, the wizard from the Hat Chooses the Wizard.
+The file contains a few Tags that define various animations that we will be using.
+[Here is the aseprite file](./sprites.aseprite).
+![alt text](wizard.png)
+
+Firstly we'll add our main collision check function on the level which takes a coordinate and returns true when a colliding tile is on that coordinate.
+
+```rust
+impl Level {
+    fn collides(&self, tile: Vector2D<i32>) -> bool {
+        if !self.bounds().contains_point(tile) {
+            return false;
+        }
+
+        let idx = (tile.x + tile.y * self.width as i32) as usize;
+
+        self.collision_map[idx / 8] & (1 << (idx % 8)) != 0
+    }
+}
+```
+
+Now for creating the player.
+This will handle the input from the player and update itself.
+Here is a potential player you could use that handles some controls but no collision or gravity.
+
+```rust
+// define a common set of number and vector type to use throughout
+type Number = Num<i32, 8>;
+type Vector = Vector2D<Number>;
+
+struct Player {
+    position: Vector,
+    velocity: Vector,
+    frame: usize,
+    flipped: bool,
+}
+
+impl Player {
+    fn new(start: Vector2D<i32>) -> Self {
+        Player {
+            position: start.change_base(),
+            velocity: (0, 0).into(),
+            frame: 0,
+            sprite: sprites::STANDING.sprite(0).into(),
+            flipped: false,
+        }
+    }
+
+    // checks whether the bottom of our sprite + 1 of the smallest subpixel values is colliding with something.
+    // if it is, then we're on the ground
+    fn is_on_ground(&self, level: &Level) -> bool {
+        level.collides(vec2(self.position.x, self.position.y + 8 + Number::from_raw(1)).floor() / 8)
+    }
+
+    // modifies the velocity accounting for various bonuses
+    fn handle_horizontal_input(&mut self, x_tri: i32, on_ground: bool) {
+        let mut x = x_tri;
+
+        // if we're trying to move in a direction opposite to what we're currently moving, we should decelerate faster.
```

Comment:
some of these comments / lines are quite long. Could you load this chapter up in the preview and break lines so you don't get a horizontal scroll bar?